scripts-available/CDB_DistType.sql

```diff
@@ -0,0 +1,138 @@
+--
+-- CDB_DistType classifies the histograms of a column into
+-- one of the basic types listed by Galtong: http://druedin.com/2012/12/08/galtungs-ajus-system/
+-- 
+-- Future improvements:
+--    variable number of bins (7 is baked in right now)
+--    catch the number of items to ensure that the sample is large enough
+--
+-- Refs:
+--    1. width_bucket/histograms: http://tapoueh.org/blog/2014/02/21-PostgreSQL-histogram
+--    2. R implementation: https://github.com/cran/agrmt
+
+CREATE OR REPLACE FUNCTION CDB_DistType ( in_array NUMERIC[] ) RETURNS text as $$
```

Comment:
The thing that makes my most disquiet here is shoving the contents of an entire column into this function as an array. This is the very definition of "not scalable", though I know the pattern is all over the place in CDB. Realistically, for a lot of these functions, we should be adding cached metadata about distribution, extents, etc, to the system rather than passing full tables scans through functions like this. There is actually already a data histogram for every indexed numeric column in the system, in the stats table, though it might not exist immediately upon import (it requires an ANALYZE to be run on the table). @rambo